dbms/src/Storages/MergeTree/MergeTreeData.cpp

```diff
@@ -1030,8 +1030,11 @@ MergeTreeData::AlterDataPartTransactionPtr MergeTreeData::alterDataPart(
             *this, part, DEFAULT_MERGE_BLOCK_SIZE, 0, 0, expression->getRequiredColumns(), ranges,
             false, nullptr, "", false, 0, DBMS_DEFAULT_BUFFER_SIZE, false);
 
+        auto compression_method = this->context.chooseCompressionMethod(
+            this->getTotalActiveSizeInBytes(),
```

Comment:
Looks incorrect. We need to get the ratio of size of data part to size of table.